### ISSUE TITLE
configure fails with syntax error

### ISSUE THREAD

**Issue by jamadden** [libuv/libuv#2862]
* **Version**: 1.38.0, upgrading from 1.36.0
* **Platform**: Travis CI, Ubuntu 16.04.6

Building gevent runs the command `pushd deps/libuv && ./autogen.sh && ./configure -C --disable-static --prefix=$BUILD_LIBS ` on Travis CI. When I try to upgrade to 1.38, [this fails](https://travis-ci.org/github/gevent/gevent/jobs/689443105):

```
checking whether make supports nested variables... (cached) yes
./configure: line 12686: syntax error near unexpected token `newline'
./configure: line 12686: `AX_PTHREAD('
```

This happens even if I use `bash` explicitly to run `configure`. 

I don't reproduce this locally on macOS.

I didn't see anything about changes to the build system in the release notes. Could I have done something wrong?

**Comment by bnoordhuis** [comment]
It's caused by #2823 but if it's breaking the build for you with no workaround, then I'll revert it.

I'm unclear on why it's failing for you though. Are you using old versions of autoconf/automake? What does `configure` around line 12686 look like on the affected system?

**Comment by jamadden** [comment]
> Are you using old versions of autoconf/automake? 

I don't know. I'm using whatever comes with Travis CI's Ubuntu 16.04 image (which is the default for all Travis builds right now). That is, unfortunately, one thing they don't specify. I can find out, though.

> What does configure around line 12686 look like on the affected system?

I can try to find out.

**Comment by jamadden** [comment]
Autoconf is version 2.69 (which matches what I have locally). Automake is 1.15 (locally I have 1.16.2).

The ten lines around 12686 are:
```shell
  AM_V=$AM_DEFAULT_VERBOSITY
  AM_DEFAULT_V=$AM_DEFAULT_VERBOSITY
fi
AM_BACKSLASH='\'


AX_PTHREAD(
    LIBS="$LIBS $PTHREAD_LIBS"
    CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
)
{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for dlopen in -ldl" >&5
```

I can see what happens if I force Travis to use Ubuntu Bionic (18.04), but that will take some time.

**Comment by jamadden** [comment]
Ah-ha. I think I figured it out. The libuv tarball includes `m4/.gitignore` which ignores all `.m4` files in that directory:
https://github.com/libuv/libuv/blob/cc51217a317e96510fbb284721d5e6bc2af31e33/m4/.gitignore#L2

But https://github.com/libuv/libuv/pull/2823 explicitly added `ax_pthread.m4` to that directory (without excluding it from being ignored). When I updated the libuv source and committed it to git, because of the .gitignore, that new file didn't get included — and because of the .gitignore, none of the tooling reported a suspicious extra file. Naturally it worked locally for me because that's where I unpacked the tarball, but a clean checkout didn't include that macro.

Perhaps that file should be listed in `m4/.gitignore`?
